core/kotlinx-coroutines-core/test/internal/LockFreeTaskQueueLinearizabilityTest.kt

```diff
@@ -14,20 +14,39 @@ import kotlin.test.*
 
 @OpGroupConfigs(OpGroupConfig(name = "consumer", nonParallel = true))
 @Param(name = "value", gen = IntGen::class, conf = "1:3")
-class LockFreeTaskQueueLinearizabilityTestSC : LockFreeTaskQueueLinearizabilityTestBase(singleConsumer = true)
+class LockFreeTaskQueueLinearizabilityTestSC : LockFreeTaskQueueLinearizabilityTestBase() {
+    private lateinit var q: LockFreeTaskQueue<Int>
+
+    @Reset
+    fun resetQueue() {
+        q = LockFreeTaskQueue(singleConsumer = true) // SINGLE-CONSUMER !!!
```

Comment:
Instead of creating a copy of the test you could create a private companion and mutate its field:

```
private companion object {
  singleConsumer = true
}

@Test
fun testMultiCosumer() {
  singleConsumer = false
  ... run test
} 
```